src/manager.rs

```diff
@@ -0,0 +1,157 @@
+use super::models::*;
+
+use piece::{Piece, PieceBag};
+use plateau::{Plateau, Player};
+use point::Point;
+
+use super::player_com::{ComError, PlayerCom, PlayerError};
+
+use std::fmt;
+
+#[derive(PartialEq)]
+pub enum Winner {
+    Player1,
+    Player2,
+    None,
+}
+
+pub struct Manager {
+    plateau: Plateau,
+    piece_bag: PieceBag,
+    player_com: PlayerCom,
+    winner: Winner,
+    p1_move_count: u32,
+    p2_move_count: u32,
+    p1_last_move: Option<String>,
+    p2_last_move: Option<String>,
+    current_piece: Option<Piece>,
+}
+
+impl Manager {
+    pub fn new(plateau: Plateau, piece_bag: PieceBag, player_com: PlayerCom) -> Manager {
```

Comment:
This is just a thought, but i wounder if we should not use some type of trait for the plateau/piecebag and rather inject it as you would an interface as apposed the actual struct?

eg. The piecebag could be expected to implement a Piece_generator trait that has a fn next() for example.